src/MeshKernel/Mesh.cpp

```diff
@@ -637,22 +637,22 @@ void Mesh::BuildTree(MeshLocations meshLocation)
     }
 }
 
-void Mesh::SearchNearestNeighbors(Point point, MeshLocations meshLocation)
+void Mesh::SearchNearestNeighbor(Point point, MeshLocations meshLocation)
 {
     BuildTree(meshLocation);
     if (meshLocation == MeshLocations::Nodes)
     {
-        m_nodesRTree.NearestNeighbors(point);
+        m_nodesRTree.NearestNeighbor(point);
```

Comment:
I think you can reduce the number of if-statements here and in other functions if you make a function that returns the RTree given the meshlocation. Then this function would become 
{
     BuildTree(meshLocation);
     GetRTree(meshLocation).NearestNeighbor(point);
}